### PR TITLE
Use URL constructor for combineURL.

### LINKED ISSUE #6754: Move URL parsing code into compatibility.js
There is combineUrl in shared/util.js which ideally suppose to use [URL constructor](https://developer.mozilla.org/en-US/docs/Web/API/URL/URL). We can add whole (or portion) of the https://github.com/Polymer/URL to the web/compatibility.js or shared/util.js (and remove for MOZCENTRAL|FIREFOX|CHROME).

### PR DESCRIPTION
**Pull request by brendandahl**
Fixes #6754

### CODE REVIEW

File: web/compatibility.js [hunk 1]
```diff
@@ -591,3 +591,620 @@ if (typeof PDFJS === 'undefined') {
...
+            break loop;
+          } else {
+            err('Code point not allowed in scheme: ' + c)
+            break loop;
+          }
+          break;
+
+        case 'scheme data':
+          if ('?' == c) {
+            query = '?';
```
Comment (xlc) [inline web/compatibility.js]: Please use the fixed version https://github.com/Polymer/URL/pull/17

**Comment by yurydelendik** [comment]
The URL polyfill at web/compatibility.js will not be accessible from core/ modules when worker is used (as is, it will regress IE10+, see also https://developer.mozilla.org/en-US/docs/Web/API/URL#Browser_compatibility). Please move that to the src/shared/util.js and use "//#if !(MOZCENTRAL || CHROME)" guard to remove that from the extension code.

File: web/compatibility.js (hunk 1 above)
Comment (yurydelendik) [inline web/compatibility.js]: yeah, this would probably not happen when "/* jshint ignore:start */" is not used.

**Comment by brendandahl** [comment]
That table says there's support in IE10 for URL. Is it not supported in the worker for IE10?

**Comment by yurydelendik** [comment]
> That table says there's support in IE10 for URL. Is it not supported in the worker for IE10?

URL.createObjectURL is supported, but not the constructor or other properties.

File: src/shared/util.js [hunk 2]
```diff
@@ -1654,6 +1632,623 @@ function loadJpegStream(id, imageUrl, objs) {
...
 
+// Polyfill from https://github.com/Polymer/URL
+/* Any copyright is dedicated to the Public Domain.
+ * http://creativecommons.org/publicdomain/zero/1.0/ */
+(function checkURLConstructor(scope) {
+  /* jshint ignore:start */
+
+  // feature detect for URL constructor
+  var hasWorkingUrl = false;
+  if (!scope.forceJURL) {
```
Comment (yurydelendik) [inline src/shared/util.js]: let's replace this with `typeof URL === 'function' && ('origin' in URL.prototype)`

File: src/shared/util.js [hunk 3]
```diff
@@ -1654,6 +1632,623 @@ function loadJpegStream(id, imageUrl, objs) {
   img.src = imageUrl;
 }
 
+// Polyfill from https://github.com/Polymer/URL
```
Comment (yurydelendik) [inline src/shared/util.js]: Can you add `//#if !(MOZCENTRAL)` preprocessor?

**Comment by yurydelendik** [comment]
r=me with comments addressed (and tests are passing). We probably need to address jslint warnings in the future if they are easy to fix.

**Pull request merged**